### PR TITLE
refactor: remove XlaOp workflow

### PR DESCRIPTION
**Pull request by polvalente**
Removes all of the XlaOp workflow
Because I had to touch the Makefile, I also took the opportunity to start speeding recompilations up.

### CODE REVIEW

File: exla/c_src/exla/exla_client.o
Comment (josevalim) [inline exla/c_src/exla/exla_client.o:1]: Is this file here intentionally? Same below.

File: exla/lib/exla/computation.ex [hunk 1]
```diff
@@ -35,43 +33,6 @@ defmodule EXLA.Computation do
...
-      client: client,
-      ref: ref,
-      output_shape: output_shape,
-      num_replicas: num_replicas,
-      num_partitions: num_partitions,
-      device_id: device_id
-    }
-  end
-
   def compile(
```
Comment (josevalim) [inline exla/lib/exla/computation.ex:36]: We should talk about this. :)

File: exla/c_src/exla/exla_client.o
Comment (polvalente) [reply exla/c_src/exla/exla_client.o:1]: No, this is a stray file from while I was working on the Makefile :)

File: exla/lib/exla/mlir/region.ex [hunk 2]
```diff
@@ -1,5 +1,5 @@
 defmodule EXLA.MLIR.Region do
-  @moduledoc false
+  @moduledoc "Holds the ref for an MLIR region."
```
Comment (josevalim) [inline exla/lib/exla/mlir/region.ex:2]: I think all MLIR modules should be false for now.

**Pull request merged**